src/server.c

```diff
@@ -4031,7 +4024,7 @@ int processCommand(client *c) {
 
     /* in case we are starting to ProcessCommand and we already have a command we assume
      * this is a reprocessing of this command, so we do not want to perform some of the actions again. */
-    int client_reprocessing_command = c->cmd ? 1 : 0;
```

Comment:
```c
    /* Now lookup the command and check ASAP about trivial error conditions
     * such as wrong arity, bad command name and so forth.
     * In case we are reprocessing a command after it was blocked,
     * we do not have to repeat the same checks */
    if (!client_reprocessing_command) {
```
Now we do a double look up even so `c->cmd` may be already set

Reply:
Yes. That is the downside

Reply:
We can check that explicitly

Reply:
Actually @uriyage I think we can limit the change as you point out. this was the original reason we handled it also in processCommand.
Let me offer a small refactor

Reply:
@enjoy-binbin @uriyage what do you guys think of the latest refactor?